### PR TITLE
Clarify how `echoMessages` should be implemented

### PR DESCRIPTION
**Pull request by lawrence-forooghian**

### CODE REVIEW

**Review by sacOO7 — CHANGES_REQUESTED** [review]

**Review by VeskeR — APPROVED** [review]

File: textile/features.textile [hunk 1]
```diff
@@ -444,7 +444,7 @@ h3(#realtime-connection). Connection
 * @(RTN1)@ @Connection@ connects to the Ably service using a "websocket":https://ably.com/topic/websockets connection. The "ably-js library":https://github.com/ably/ably-js supports additional transports such as Comet and XHR streaming; however non-browser client libraries typically use only a websocket transport
 * @(RTN2)@ The default host used for realtime "websocket":https://ably.com/topic/websockets connections is @realtime.ably.io@, and the following query string params should be used when opening a new connection:
 ** @(RTN2a)@ @format@ should be @msgpack@ (default) or @json@
-** @(RTN2b)@ @echo@ should be @true@ by default; @false@ will prevent messages published by the client being echoed back
+** @(RTN2b)@ @echo@ should be @true@ if the @echoMessages@ client option is true; else it should be @false@, which will prevent messages published by the client being echoed back
```
[the remark this answers is not in the document]
Comment (maratal) [reply textile/features.textile:447]: I agree with @sacOO7, this sentence can be simplified.

[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply textile/features.textile:447]: But the `echoMessages` client option is a boolean, and the `echo` connection parameter is a string, right? It’s not obvious that `true` maps to `"true"` and `false` maps to `"false"`.

[the remark this answers is not in the document]
Comment (sacOO7) [reply textile/features.textile:447]: Okay, got it

**Review by sacOO7 — APPROVED** [review]
Lgtm

**Comment by SimonWoolf** [comment]
@lawrence-forooghian what was the purpose of this change?

And this isn't exactly a clarification, it's a material change in behaviour -- before this pr a lib only needs to set the echo qs param if the user specifies echoMessages=false, after this pr it must set it always?

**Comment by lawrence-forooghian** [comment]
The intention of this change was to make it clear what a client needed to do in order to implement the behaviour described in RTC1a: "If false, it prevents messages originating from this connection being echoed back on the same connection". It does this by stating that this is achieved by using the `echo` connection query param.

> it's a material change in behaviour -- before this pr a lib only needs to set the echo qs param if the user specifies echoMessages=false, after this pr it must set it always?

I don't agree — even before this, it still said "`echo` should be `true` by default" i.e. it still implied that the client was always meant to send `echo`.

But yes, if `true` is the _server_ default then I agree with #465; I'll do a PR to address.

**Comment by SimonWoolf** [comment]
> even before this, it still said "echo should be true by default"

Fair point, that item was very poorly phrased -- it's one of those very early ones which were terrible for confusing SDK implementation requirements with expected overall system behaviour

**Pull request merged**